src/hooks/useGetPurposeMessageKeys.ts

```diff
@@ -4,34 +4,49 @@ import { useMemo } from 'preact/hooks';
 
 import { DefinedMessage } from '@transcend-io/internationalization';
 
-const CUSTOM_PURPOSE_MESSAGE_PREFIX = 'cm-ui.purpose';
+const PURPOSE_MESSAGE_PREFIX = 'purpose.trackingType';
 
 export const useGetPurposeMessageKeys = ({
   consentSelection,
   defaultPurposeToMessageKey,
+  inverted = false,
 }: {
   /** The configured airgap purpose types */
   consentSelection: ConsentSelection;
   /** The lookup of messages for default purpose types */
   defaultPurposeToMessageKey: Record<string, DefinedMessage>;
+  /** Whether messages are for CompleteOptionsInverted */
+  inverted?: boolean;
 }): Record<string, DefinedMessage> => {
   const purposeToMessageKey: Record<string, DefinedMessage> = useMemo(
     () =>
       // the purpose type is unique for the bundle
-      Object.keys(consentSelection ?? {}).reduce((allMessages, purposeType) => {
-        if (allMessages[purposeType]) {
-          return allMessages;
-        }
-        const customPurposeMessageLabel = `${CUSTOM_PURPOSE_MESSAGE_PREFIX}.${purposeType}`;
-        return {
-          ...allMessages,
-          [purposeType]: {
-            id: customPurposeMessageLabel,
-            defaultMessage: purposeType,
-            description: `Translatable name for custom purpose '${purposeType}'`,
-          } as DefinedMessage,
-        };
-      }, defaultPurposeToMessageKey as Record<string, DefinedMessage>),
+      [...Object.keys(consentSelection ?? {}), 'Essential'].reduce(
+        (allMessages, purposeType) => {
+          if (!inverted) {
+            const purposeMessageLabel = `${PURPOSE_MESSAGE_PREFIX}.${purposeType}.title`;
+            return {
+              ...allMessages,
+              [purposeType]: {
+                id: purposeMessageLabel,
+                defaultMessage: purposeType,
+                description: `Translatable name for purpose '${purposeType}'`,
+              } as DefinedMessage,
+            };
+          }
+          return {
+            ...allMessages,
+            [purposeType]: {
+              id: `cm-ui.purpose.${purposeType}`,
```

Comment:
we shouldnt have any reference to the cm-ui prefix in the updated package. i dont think this inverted parameter is necessary. afaict it should just be the L7 change, then adding Essential on L24

Reply:
oh actually i see whats going on here, nvm, we will have to split out inverted. lets make a `useGetInvertedPurposeMessageKeys` hook for now (in the future these should probably also be messages?)